packages/preview-middleware-client/src/adp/controllers/BaseDialog.controller.ts

```diff
@@ -41,6 +43,53 @@ export default abstract class BaseDialog extends Controller {
 
     abstract buildDialogData(): Promise<void>;
 
+    /**
+     * Handles fragment name input change
+     *
+     * @param event Event
+     */
+    onFragmentNameInputChange(event: Event) {
+        const source = event.getSource<Input>();
+
+        const fragmentName: string = source.getValue().trim();
+        const fragmentList: { fragmentName: string }[] = this.model.getProperty('/fragmentList');
+
+        if (fragmentName.length <= 0) {
+            this.dialog.getBeginButton().setEnabled(false);
+            source.setValueState(ValueState.None);
+            this.model.setProperty('/newFragmentName', null);
+        } else {
+            const fileExists = fragmentList.find((f: { fragmentName: string }) => {
+                return f.fragmentName === `${fragmentName}.fragment.xml`;
+            });
+
+            const isValidName = /^[a-zA-Z_][a-zA-Z0-9_-]*$/.test(fragmentName);
+
+            if (fileExists) {
+                source.setValueState(ValueState.Error);
+                source.setValueStateText(
+                    'Enter a different name. The fragment name that you entered already exists in your project.'
+                );
+                this.dialog.getBeginButton().setEnabled(false);
+            } else if (!isValidName) {
+                source.setValueState(ValueState.Error);
+                source.setValueStateText('A Fragment Name cannot contain white spaces or special characters.');
+                this.dialog.getBeginButton().setEnabled(false);
+            } else {
+                this.dialog.getBeginButton().setEnabled(true);
+                source.setValueState(ValueState.None);
+                this.model.setProperty('/newFragmentName', fragmentName);
+            }
+        }
+    }
+
+    setEscapeHandler() {
```

Comment:
Please add JSDoc

Reply:
Added in: https://github.com/SAP/open-ux-tools/pull/1378/commits/38e7284c4c05f224d4b0a45bc40f8e6b93ddffe2

Reply:
thanks